igvc/src/global_mapper/main.cpp

```diff
@@ -11,56 +11,93 @@
 #include <stdlib.h>
 #include <string>
 
-pcl::PointCloud<pcl::PointXYZ>::Ptr global_map;
+pcl::PointCloud<pcl::PointXYZRGB>::Ptr global_map;
 ros::Publisher _pointcloud_pub;
-ros::Publisher _pointcloud_incremental_pub;
+ros::Publisher _pointcloud_prob_pub;
 tf::TransformListener *tf_listener;
-std::set<std::string> frames_seen;
 bool firstFrame;
 double maxCorrDist;
 int maxIter;
-pcl::octree::OctreePointCloudSearch<pcl::PointXYZ> octree(.01f);
+pcl::octree::OctreePointCloudSearch<pcl::PointXYZRGB>::Ptr octree;
 double searchRadius;
+double octree_resolution;
+double probability_thresh;
 
-void icp_transform(pcl::PointCloud<pcl::PointXYZ>::Ptr input)
+//Probabilistic mapping
+//std::vector<pcl::PointCloud<pcl::PointXYZRGB>::Ptr> pointcloud_list;
+std::list<pcl::PointCloud<pcl::PointXYZRGB>::Ptr> pointcloud_list;
+int queue_window_size;
+
+void icp_transform(pcl::PointCloud<pcl::PointXYZRGB>::Ptr input)
 {
-  if (!firstFrame)
+  if (pointcloud_list.size() < queue_window_size)
+  {
+    pointcloud_list.push_back(input);
+  } 
+  if (pointcloud_list.size() == queue_window_size)
   {
-    pcl::IterativeClosestPoint<pcl::PointXYZ, pcl::PointXYZ> icp;
-    icp.setMaxCorrespondenceDistance(maxCorrDist);
-    icp.setMaximumIterations(maxIter);
-    icp.setInputSource(input);
-    icp.setInputTarget(global_map);
-    pcl::PointCloud<pcl::PointXYZ> Final;
-    pcl::PointCloud<pcl::PointXYZ> new_points;
-    icp.align(Final);
-    for (unsigned int i = 0; i < Final.size(); ++i)
+    //Probabilistic calculations here
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr local_pc = pcl::PointCloud<pcl::PointXYZRGB>::Ptr(new pcl::PointCloud<pcl::PointXYZRGB>());
+    pcl::octree::OctreePointCloudSearch<pcl::PointXYZRGB> local_octree(octree_resolution);
+
+    local_octree.setInputCloud(local_pc);
+
+    std::list<pcl::PointCloud<pcl::PointXYZRGB>::Ptr>::iterator it = pointcloud_list.begin();
+    for (unsigned int i = 0; i < pointcloud_list.size(); i++)
```

Comment:
Since you are already using an iterator and never using i, I would recommend.

`for(auto it = pointcloud_list.begin(); it != pointcloud_list.end(); it++)`